src/Microsoft.Health.Fhir.Core/Features/Operations/Import/GetImportRequestHandler.cs

```diff
@@ -45,93 +45,90 @@ public async Task<GetImportResponse> Handle(GetImportRequest request, Cancellati
-        }
 
-        private async Task<(List<ImportOperationOutcome> completedOperationOutcome, List<ImportFailedOperationOutcome> failedOperationOutcome)> GetProcessingResultAsync(long groupId, CancellationToken cancellationToken)
-        {
-            var start = Stopwatch.StartNew();
-            var jobs = await _queueClient.GetJobByGroupIdAsync(QueueType.Import, groupId, true, cancellationToken);
-            var duration = start.Elapsed.TotalSeconds;
-            var completedOperationOutcome = new List<ImportOperationOutcome>();
-            var failedOperationOutcome = new List<ImportFailedOperationOutcome>();
-            foreach (var job in jobs.Where(_ => _.Id != groupId && _.Status == JobStatus.Completed)) // ignore coordinator && not completed
+            static (List<ImportOperationOutcome> Completed, List<ImportFailedOperationOutcome> Failed) GetProcessingResultAsync(IList<JobInfo> jobs)
             {
-                var definition = JsonConvert.DeserializeObject<ImportProcessingJobDefinition>(job.Definition);
-                var result = JsonConvert.DeserializeObject<ImportProcessingJobResult>(job.Result);
-                completedOperationOutcome.Add(new ImportOperationOutcome() { Type = definition.ResourceType, Count = result.SucceededResources, InputUrl = new Uri(definition.ResourceLocation) });
-                if (result.FailedResources > 0)
+                var completed = new List<ImportOperationOutcome>();
+                var failed = new List<ImportFailedOperationOutcome>();
+                foreach (var job in jobs.Where(_ => _.Status == JobStatus.Completed))
                 {
-                    failedOperationOutcome.Add(new ImportFailedOperationOutcome() { Type = definition.ResourceType, Count = result.FailedResources, InputUrl = new Uri(definition.ResourceLocation), Url = result.ErrorLogLocation });
+                    var definition = JsonConvert.DeserializeObject<ImportProcessingJobDefinition>(job.Definition);
+                    var result = JsonConvert.DeserializeObject<ImportProcessingJobResult>(job.Result);
+                    completed.Add(new ImportOperationOutcome() { Type = definition.ResourceType, Count = result.SucceededResources, InputUrl = new Uri(definition.ResourceLocation) });
+                    if (result.FailedResources > 0)
+                    {
+                        failed.Add(new ImportFailedOperationOutcome() { Type = definition.ResourceType, Count = result.FailedResources, InputUrl = new Uri(definition.ResourceLocation), Url = result.ErrorLogLocation });
+                    }
                 }
-            }
 
-            await Task.Delay(TimeSpan.FromSeconds(duration * 10), cancellationToken); // throttle to avoid misuse.
+                // group success results by url
+                var groupped = completed.GroupBy(o => o.InputUrl).Select(g => new ImportOperationOutcome() { Type = g.First().Type, Count = g.Sum(_ => _.Count), InputUrl = g.Key }).ToList();
```

Comment:
This is also a behavior change we need to verify with @EXPEkesheth. Currently there is a single entry returned per processing job.

Reply:
We never made any promises on number of worker jobs created for a single input file. This output will look like it is a single job per file. I don't think we need to notify customer about the change.